### ISSUE TITLE
unexpected code generated with class expression

### ISSUE THREAD

**Issue by vvakame** [Microsoft/TypeScript#4630]
tested on 690b87c2097b0101969c941f085b1f660ecc1a7e

source code.
```
new class {
	hi() {
		return "Hi!";
	}
}().hi();
```

expected.
```
new ((function () {
    function class_1() {
    }
    class_1.prototype.hi = function () {
        return "Hi!";
    };
    return class_1;
})())().hi();
```

actual.
```
new (function () {
    function class_1() {
    }
    class_1.prototype.hi = function () {
        return "Hi!";
    };
    return class_1;
})()().hi();
```

```
$ tsc --target es5 test.ts
$ node test.js
/private/tmp/tsc-1.6/test.js:8
})()().hi();
      ^

TypeError: Cannot read property 'hi' of undefined
    at Object.<anonymous> (/private/tmp/tsc-1.6/test.js:8:7)
    at Module._compile (module.js:430:26)
    at Object.Module._extensions..js (module.js:448:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:471:10)
    at startup (node.js:117:18)
    at node.js:951:3
```

**Comment by mhegazy** [comment]
that is similar to function expressions (#4472). i will need to double check the ES6 grammar to be sure though.

**Comment by mhegazy** [comment]
@DanielRosenwasser can you take a look.

**Comment by vladima** [comment]
absence of parens effective make generated code looks like this:
```javascript
var x = new (function () {
    function class_1() {
    }
    class_1.prototype.hi = function () {
        return "Hi!";
    };
    return class_1;
})()

x().hi();
```

**Comment by mhegazy** [comment]
my bad. we should parenthesize the output.

**Comment by DanielRosenwasser** [comment]
Should be fixed for 1.8. Thanks for the heads up @vvakame
